### PR TITLE
fix(TDI-46955): TCK - Outline - Flow variable

### PR DESCRIPTION
**Pull request by wwang-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-46955

**What is the new behavior?**
https://jira.talendforge.org/browse/TDI-46955

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by ozhelezniak-talend — COMMENTED** [review]
Left a comment about constant name and code duplication. In general looks good.

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ComponentModel.java [hunk 1]
```diff
@@ -358,6 +359,25 @@ public List<? extends INodeReturn> createReturns(final INode node) {
                 returnVariables.add(returnNode);
             }
         }
+        
+        if (detail.getMetadata().containsKey(TaCoKitConst.META_KEY_FLOW_VARIABLE)) {
+            String flowVariableMetaValue = detail.getMetadata().getOrDefault(TaCoKitConst.META_KEY_FLOW_VARIABLE, "");
+            for (String string : flowVariableMetaValue.split(TaCoKitConst.AFTER_VARIABLE_LINE_DELIMITER)) {
+                String[] split = string.split(TaCoKitConst.AFTER_VARIABLE_VALUE_DELIMITER);
+                String key = split[0];
```
[the remark this answers is not in the document]
Comment (wwang-talend) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ComponentModel.java:367]: totally agree, think we need to make AFTER_VARIABLE Deprecated, and use a var like RETURN_VARIABLE, as that only for return var.

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ComponentModel.java [hunk 2]
```diff
@@ -358,6 +359,25 @@ public List<? extends INodeReturn> createReturns(final INode node) {
                 returnVariables.add(returnNode);
             }
         }
+        
+        if (detail.getMetadata().containsKey(TaCoKitConst.META_KEY_FLOW_VARIABLE)) {
+            String flowVariableMetaValue = detail.getMetadata().getOrDefault(TaCoKitConst.META_KEY_FLOW_VARIABLE, "");
+            for (String string : flowVariableMetaValue.split(TaCoKitConst.AFTER_VARIABLE_LINE_DELIMITER)) {
+                String[] split = string.split(TaCoKitConst.AFTER_VARIABLE_VALUE_DELIMITER);
```
[the remark this answers is not in the document]
Comment (wwang-talend) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ComponentModel.java:366]: how about RETURN_VARIABLE ?

[the remark this answers is not in the document]
Comment (ozhelezniak-talend) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ComponentModel.java:366]: why not. ok to me

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ComponentModel.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ozhelezniak-talend) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ComponentModel.java:367]: What about code duplication?
Even if we make AfterVariable a deprecated. we should leave it for a few releases to be at that state. Maybe to put a warning when we think to remove it. So this code will be here and we'll need to maintain it for some time.

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ComponentModel.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ozhelezniak-talend) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ComponentModel.java:366]: Yes, why not. It's ok to me.

**Review by ozhelezniak-talend — APPROVED** [review]
LGTM

**Review by undx — APPROVED** [review]

**Pull request merged**